nesis/api/core/document_loaders/sharepoint.py

```diff
@@ -23,158 +27,278 @@
-                                )
-                                continue
-                            rag_metadata: dict = document.rag_metadata
-                            if rag_metadata is None:
-                                continue
-                            for document_data in rag_metadata.get("data") or []:
-                                try:
-                                    http_client.delete(
-                                        url=f"{rag_endpoint}/v1/ingest/{document_data['doc_id']}"
-                                    )
-                                except:
-                                    _LOG.warn(
-                                        f"Failed to delete document {document_data['doc_id']}"
-                                    )
-
-                            try:
-                                delete_document(document_id=file.unique_id)
-                            except:
-                                _LOG.warn(
-                                    f"Failed to delete file {file.name}'s record. Continuing anyway..."
-                                )
-
-                    response = http_client.upload(
-                        url=f"{rag_endpoint}/v1/ingest",
-                        filepath=downloaded_file_name,
-                        field="file",
-                    )
-                    response_json = json.loads(response)
-
-                    save_document(
-                        document_id=file.unique_id,
-                        filename=file.name,
-                        base_uri=site_url,
-                        rag_metadata=response_json,
-                        store_metadata={
-                            "file_url": f"{location_name}/{file.name}",
-                            "file_name": file.name,
-                            "etag": file.unique_id,
-                            "size": file.length,
-                            "author": file.author,
-                            "last_modified": file.time_last_modified.strftime(
-                                DEFAULT_DATETIME_FORMAT
-                            ),
-                            "version_id": f"{file.major_version}-{file.minor_version}",
-                        },
-                    )
+    except Exception as file_ex:
+        _LOG.exception(
+            f"Error fetching and updating documents - Error: {file_ex}", exc_info=True
+        )
 
-                    _LOG.info(
-                        f"Done syncing file {file.name} in location {location_name}"
-                    )
-                except Exception as ex:
-                    _LOG.warn(
-                        f"Error when getting and ingesting file {file.name} - {ex}"
-                    )
 
-        _LOG.info(f"Completed syncing to endpoint {rag_endpoint}")
+def _process_file(file, connection, rag_endpoint, http_client, metadata, cache_client):
+    self_link = file.serverRelativeUrl
```

Comment:
@akizito the `self_link` needs to be the full url. This is to guarantee its uniqueness. It looks like `EncodedAbsUrl` may have the answer.